digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java

```diff
@@ -110,24 +118,27 @@ public void run(TaskRequest request)
     {
         long taskId = request.getTaskId();
 
-        // set task name to thread name so that logger shows it
-        try (SetThreadName threadName = new SetThreadName(request.getTaskName())) {
-            try (TaskLogger taskLogger = callback.newTaskLogger(request)) {
-                TaskContextLogging.enter(LogLevel.DEBUG, taskLogger);
-                try {
-                    runningTaskMap.put(taskId, request);
+        tm.begin(() -> {
```

Comment:
I think the intention there is that `ScheduleTime.runNow(instant)` should be replaced by something because it is not deterministic. It's not very important....